controller/instance_manager_controller.go

```diff
@@ -1702,34 +1707,20 @@ func (imc *InstanceManagerController) deleteOrphans(im *longhorn.InstanceManager
 	return nil
 }
 
-func (imc *InstanceManagerController) isOrphanDeletable(orphan *longhorn.Orphan, autoDelete bool, instanceMap instanceProcessMap, isInstanceScheduledOnNode func(node, instance string) (bool, error)) (isDeletable bool, err error) {
-	if autoDelete {
-		return true, nil
-	}
-
-	instanceName := orphan.Spec.Parameters[longhorn.OrphanInstanceName]
-	instanceProc, instanceExist := instanceMap[instanceName]
-	instanceCRScheduledBack, err := isInstanceScheduledOnNode(orphan.Spec.NodeID, instanceName)
-	if err != nil {
-		return false, errors.Wrapf(err, "failed to check CR scheduled node for %v orphan %v", orphan.Spec.Type, instanceName)
-	}
-
-	var instanceState longhorn.InstanceState
-	if instanceExist {
-		instanceState = instanceProc.Status.State
-	} else {
-		instanceState = longhorn.InstanceStateTerminated
+func (imc *InstanceManagerController) deleteOrphan(orphan *longhorn.Orphan) error {
+	imc.logger.Infof("Deleting Orphan %q", orphan.Name)
+	err := imc.ds.DeleteOrphan(orphan.Name)
+	switch {
+	case datastore.ErrorIsNotFound(err):
+		return nil
+	case err != nil:
+		return errors.Wrapf(err, "failed to delete orphan %v", orphan.Name)
 	}
```

Comment:
Use if statement for readability.
```suggestion
	if err := imc.ds.DeleteOrphan(orphan.Name); err != nil {
		if datastore.ErrorIsNotFound(err) {
			return nil
		}
		return errors.Wrapf(err, "failed to delete orphan %q", orphan.Name)
	}
```